src/core/tiledmesh/qgscesiumutils.h

```diff
@@ -21,12 +21,96 @@
 
 #include "qgis_core.h"
 #include "qgsbox3d.h"
+#include "qgis_sip.h"
 #include "nlohmann/json_fwd.hpp"
 
 #ifndef SIP_RUN
 using namespace nlohmann;
 #endif
 
+class QgsSphere;
+
+/**
+ * \brief Represents a oriented (rotated) bounding box in 3 dimensions.
+ *
+ * \ingroup core
+ *
+ * \warning Non-stable API, exposed to Python for unit testing only.
+ *
+ * \since QGIS 3.34
+ */
+class CORE_EXPORT QgsOrientedBoundingBox
```

Comment:
It would be good to make this class more generic like QgsSphere, because OBBs are used in other protocols too... and move cesium-specific JSON parsing out of this class...? We could also call it e.g. QgsOrientedBox3d to make it clear it is 3D box, and use naming convention similar to QgsBox3d.

(By the way, we should settle on 3D vs 3d in class names, the mixture we have now is killing me :laughing: )

Reply:
Can we defer that? I've marked this class as non stable because I'm quite confident the API needs will get refined over time (as opposed to the sphere class, which is a very straightforward API to define). Eg in its current form it's very cesium specific wrt to the handling of half axes - this might not be the best approach for a generic class though.

Reply:
Actually, I've gone ahead and done that. But I've left it tagged as non-stable api for now.